### ISSUE TITLE
RequestStatsMessage; local class incompatible: stream classdesc serialVersionUID = `X`, local class serialVersionUID = `Y`

### ISSUE THREAD

**Issue by pierDipi** [openshift-knative/eventing-hyperfoil-benchmark#29]
I'm seeing these errors in the `hyperfoil-cluster-controller`
```
17:39:18,366 ERROR (vert.x-eventloop-thread-1) [i.h.c.ControllerVerticle] Unknown message type: null
java.io.InvalidClassException: io.hyperfoil.clustering.messages.RequestStatsMessage; local class incompatible: stream classdesc serialVersionUID = 7013716813918255356, local class serialVersionUID = 4091445394494914523
	at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:689)
	at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2014)
	at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1864)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2195)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at io.hyperfoil.clustering.messages.ObjectCodec.decodeFromWire(ObjectCodec.java:73)
	at io.vertx.core.eventbus.impl.clustered.ClusteredMessage.decodeBody(ClusteredMessage.java:195)
	at io.vertx.core.eventbus.impl.clustered.ClusteredMessage.body(ClusteredMessage.java:100)
	at io.hyperfoil.clustering.ControllerVerticle.lambda$start$10(ControllerVerticle.java:170)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:50)
	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:168)
	at io.vertx.core.eventbus.impl.MessageConsumerImpl.dispatch(MessageConsumerImpl.java:187)
	at io.vertx.core.eventbus.impl.HandlerRegistration$InboundDeliveryContext.next(HandlerRegistration.java:162)
	at io.vertx.core.eventbus.impl.HandlerRegistration$InboundDeliveryContext.dispatch(HandlerRegistration.java:127)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:111)
	at io.vertx.core.eventbus.impl.HandlerRegistration.dispatch(HandlerRegistration.java:103)
	at io.vertx.core.eventbus.impl.MessageConsumerImpl.deliver(MessageConsumerImpl.java:193)
	at io.vertx.core.eventbus.impl.MessageConsumerImpl.doReceive(MessageConsumerImpl.java:178)
	at io.vertx.core.eventbus.impl.HandlerRegistration.lambda$receive$0(HandlerRegistration.java:54)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

How to reproduce:
- download PR https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27 (it might be reproducible from main too)
- run `export TEST_CASE=tests/broker/kafka/p10-r3-unordered/`
- run `./bin/run_upstream_nightly.sh`
- once the benchmark is started, follow logs `oc logs -n hyperfoil hyperfoil-cluster-controller -f`

**Comment by pierDipi** [comment]
This happens after we sent stats over the event bus [1] and the controller is trying to decode the object.


[1]
https://github.com/openshift-knative/eventing-hyperfoil-benchmark/blob/0d1742b86bbb4609e9d769b242f16760b43737c5/vertx-receiver/src/main/java/io/openshift/serverless/knative/eventing/hyperfoil/benchmark/VertxReceiverVerticle.java#L104

**Comment by willr3** [comment]
I am getting `error: unrecognized condition: "jsonpath={.status.state}=AtLatestKnown"` in the console from `./bin/run_upstream_nightly.sh`

**Comment by pierDipi** [comment]
`oc` version 4.10 is required for that feature

**Comment by willr3** [comment]
does that require ocp to be 4.10 as well? our cluster is currently 4.9.7

**Comment by willr3** [comment]
```
Applying installation/manifests/nightly/000-subscription-hyperfoil.yaml
subscription.operators.coreos.com/hyperfoil-bundle created
error: state is not found
Creating namespaces
namespace/kafka configured
Applying installation/manifests/nightly/000-subscription-hyperfoil.yaml
subscription.operators.coreos.com/hyperfoil-bundle unchanged
error: state is not found
```

**Comment by pierDipi** [comment]
That error should be retried in https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27/files#diff-00a80c5821edea2ebf676056aa4c9a24e57379ef52cefecb2ccffaaf4cc362c9R27, can you run it again?

I think to improve this we can:
- add a sleep here https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27/files#diff-eacd8235d96534a6fd7e0d5a342f97d9179a4273041c993f2f795033dcb6b3ceR41-R42 between applying and waiting to be running so that we wait for the resource to be reconciled
- IMO that oc wait with `jsonpath` shouldn't fail when a field is not found (I'll open an issue in k8s).

**Comment by willr3** [comment]
I am running @ 98b0403 and the error is being logged twice so I think it is running twice.

**Comment by pierDipi** [comment]
Ok, can you try with the sleep https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27/commits/07500b17d37ad8c06bca224788265d34785ab7b2 ?

**Comment by willr3** [comment]
@ 07500b1 and I still get two `state is not found` when running in the lab
```
Creating namespaces
namespace/kafka created
Applying installation/manifests/nightly/000-subscription-hyperfoil.yaml
subscription.operators.coreos.com/hyperfoil-bundle created
error: state is not found
Creating namespaces
namespace/kafka configured
Applying installation/manifests/nightly/000-subscription-hyperfoil.yaml
subscription.operators.coreos.com/hyperfoil-bundle unchanged
error: state is not found
Deleting installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-sink.yaml
```

**Comment by pierDipi** [comment]
Ok, if you apply manually 
```
oc apply -f installation/manifests/nightly/000-subscription-hyperfoil.yaml
```
does Hyperfoil operator starts? 

maybe in the lab we already have hyperfoil and so that won't work, so we need to exclude the hyperfoil subscription when running in the lab.

**Comment by willr3** [comment]
manually applying it creates the subscription then `oc get ClusterServiceVersion hyperfoil-operator.v0.15.1 -n openshift-operators -o jsonpath='{.status.phase}'` shows `Succeeded` and I was able to create a Hyperfoil instance in the `hyperfoil` namespace

**Comment by willr3** [comment]
what version of openshift were you using for the tests?

**Comment by pierDipi** [comment]
4.9 and 4.10.

The problem is that reconciling resources to reach the ready state takes time so we need to wait or retry more times when we call [`wait_for_operators_to_be_running`](https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27/files#diff-eacd8235d96534a6fd7e0d5a342f97d9179a4273041c993f2f795033dcb6b3ceR43).

**Comment by pierDipi** [comment]
> 4.9 and 4.10.

That's the other wait condition https://github.com/openshift-knative/eventing-hyperfoil-benchmark/pull/27/files#diff-eacd8235d96534a6fd7e0d5a342f97d9179a4273041c993f2f795033dcb6b3ceR123-R126

**Comment by willr3** [comment]
I triggered an update from `4.9.7` to `4.9.26` but on `4.9.7` the hyperfoil-bundle subscription never had a `status.state` field when running `oc -n openshift-operators get subscription.operators.coreos.com/hyperfoil-bundle -o json`

**Comment by pierDipi** [comment]
I'm not sure why, this is what I get on 4.9
```shell
pierdipi@pierdipi eventing-hyperfoil-benchmark (performance-testing-improvement-plan) $ ./bin/run_upstream_nightly.sh 
HYPERFOIL_SERVER_URL: <redacted>
KNATIVE_MANIFESTS: installation/manifests/nightly/000-subscription-hyperfoil.yaml,installation/manifests/nightly/100-hyperfoil.yaml,installation/manifests/nightly/000-subscription-amq-streams.yaml,installation/manifests/nightly/100-kafka.yaml,installation/manifests/nightly/eventing/eventing-core.yaml,installation/manifests/nightly/eventing/mt-channel-broker.yaml,installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-controller.yaml,installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-broker.yaml,installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-source.yaml,installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-channel.yaml,installation/manifests/nightly/eventing-kafka-broker/eventing-kafka-sink.yaml
TEST_CASE: tests/broker/kafka/p10-r3-unordered/
TEST_CASE_NAMESPACE: perf-test
WORKER_ONE: node-role.kubernetes.io/worker=
Creating namespaces
namespace/kafka created
Applying installation/manifests/nightly/000-subscription-hyperfoil.yaml
subscription.operators.coreos.com/hyperfoil-bundle created
subscription.operators.coreos.com/hyperfoil-bundle condition met
clusterserviceversion.operators.coreos.com/hyperfoil-operator.v0.15.1 condition met
Applying installation/manifests/nightly/100-hyperfoil.yaml
namespace/hyperfoil created
hyperfoil.hyperfoil.io/hyperfoil-cluster created
subscription.operators.coreos.com/hyperfoil-bundle condition met
clusterserviceversion.operators.coreos.com/hyperfoil-operator.v0.15.1 condition met
Applying installation/manifests/nightly/000-subscription-amq-streams.yaml
subscription.operators.coreos.com/amq-streams created
subscription.operators.coreos.com/amq-streams condition met
subscription.operators.coreos.com/hyperfoil-bundle condition met
clusterserviceversion.operators.coreos.com/amqstreams.v2.1.0-0 condition met
clusterserviceversion.operators.coreos.com/hyperfoil-operator.v0.15.1 condition met
Applying installation/manifests/nightly/100-kafka.yaml
kafka.kafka.strimzi.io/my-cluster created
configmap/kafka-metrics created

# etc ...
```

Versions:
```
Client Version: 4.10.5
Server Version: 4.9.0-0.nightly-2022-04-14-125839
Kubernetes Version: v1.22.8+1601aad
```

**Comment by pierDipi** [comment]
Fixed by upgrading to 0.19 libraries.

**Comment by pierDipi** [comment]
@willr3 perhaps for the problem in the lab we can track it in a separate issue?
